### ISSUE TITLE
Compiling with absolute paths cause an infinite loop

### ISSUE THREAD

**Issue by Chobbes** [aantron/bisect_ppx#180]
Here's a silly test case...

Put this in a file in some directory called `test.ml`

```
let main () = print_string "Hello, world!\n";;                                                                                   

main ()
```

This works:

```
ocamlfind ocamlopt -linkpkg -package bisect_ppx test.ml -o test
rm -f bisect0001.out &&  ./test
bisect-ppx-report -ignore-missing-files -verbose -I . -html coverage/ bisect0001.out
```

However, if you use an absolute path like so, it causes `bisect-ppx-report` to loop forever, pegging the CPU at 100%.

```
ocamlfind ocamlopt -linkpkg -package bisect_ppx $PWD/test.ml -o test
rm -f bisect0001.out &&  ./test
bisect-ppx-report -ignore-missing-files -verbose -I . -html coverage/ bisect0001.out
```

In case it is relevant, here is my `opam list`

```
# Packages matching: installed
# Name                  # Installed       # Synopsis
afl                     2.52b             American Fuzzy Lop fuzzer by Michal Zalewski, repackaged for convenient use in opam.
afl-persistent          1.2               use afl-fuzz in persistent mode
base-bigarray           base
base-num                base              Num library distributed with the OCaml compiler
base-threads            base
base-unix               base
biniou                  1.2.0             Binary data format designed for speed, safety, ease of use and backward compatibility as
bisect_ppx              1.3.4             Code coverage for OCaml
camlp5                  7.06.10-g84ce6cc4 Preprocessor-pretty-printer of OCaml
cmdliner                1.0.2             Declarative definition of command line interfaces for OCaml
conf-gmp                1                 Virtual package relying on a GMP lib system installation
conf-m4                 1                 Virtual package relying on m4
conf-perl               1                 Virtual package relying on perl
conf-python-2-7         1.0               Virtual package relying on Python-2.7 installation
conf-which              1                 Virtual package relying on which
coq                     8.8.2             pinned to version 8.8.2
coq-ext-lib             0.9.8             A basic library of basic Coq datatypes, definitions, theorems, and
coq-mathcomp-ssreflect  1.7.0             Small Scale Reflection
coq-simple-io           0.2               IO monad for Coq
cppo                    1.6.5             Equivalent of the C preprocessor for OCaml programs
dune                    1.4.0             Fast, portable and opinionated build system
easy-format             1.3.1             High-level and functional interface to the Format module of the OCaml standard library
jbuilder                transition        This is a transition package, jbuilder is now named dune. Use the dune
menhir                  20181026          An LR(1) parser generator
num                     0                 The Num library for arbitrary-precision integer and rational arithmetic
ocaml                   4.04.0            The OCaml compiler (virtual package)
ocaml-config            1                 OCaml Switch Configuration
ocaml-migrate-parsetree 1.1.0
ocaml-variants          4.04.0+afl        OCaml 4.04.0, with afl-fuzz instrumentation backported from 4.05
ocamlbuild              0.12.0            OCamlbuild is a build system with builtin rules to easily build most OCaml projects.
ocamlfind               1.8.0             A library manager for OCaml
pprint                  20180528          A pretty-printing combinator library and rendering engine.
ppx_tools_versioned     5.2.1             A variant of ppx_tools based on ocaml-migrate-parsetree
re                      1.7.3             RE is a regular expression library for OCaml
result                  1.3               Compatibility Result module
topkg                   1.0.0             The transitory OCaml software packager
user-setup              0.7               Helper for the configuration of editors for the use of OCaml tools
yojson                  1.4.1             Yojson is an optimized parsing and printing library for the JSON format
z3                      4.7.1             Z3 solver
zarith                  1.7               Implements arithmetic and logical operations over arbitrary-precision integers
```

**Comment by Chobbes** [comment]
For my purposes I'm trying to use Bisect to generate coverage reports for QuickChick, which is a testing tool that extracts Coq code to ocaml code to run it. QuickChick extracts ocaml code to an essentially random directory, so it's pretty helpful to be able to compile with absolute paths, and it's unexpected that this wouldn't work because the program still compiles and runs fine. Bisect just chokes. For this reason I would be much happier with the first option.

Additionally, I think supporting absolute paths would make Bisect easier for people to shoehorn in to more projects.

But yes, it definitely should not go into an infinite loop :). I was very confused for a while!

**Comment by aantron** [comment]
@Chobbes Agreed.

Is the directory in which you are starting Bisect at the "root" of the "project" that the coverage report is being generated for?

If yes, we can use the cwd to compute the relative path from the absolute path.

If no, we probably need to add a command-line parameter for setting the project root directory, for the purpose of generating the report.

I guess another option is to avoid needing a root directory, and simply treat `/` as the root. We would then have long paths in the report, but that's probably not an issue. Now that I think of it, maybe that's the right way, and everything will "just work" if that infinite loop is simply solved.

I'll experiment with different ways to fix this shortly.

**Comment by aantron** [comment]
@Chobbes, the attached commit fixes the issue as far as I can tell. Please let me know if that is so, and if there are any further problems!

Are you comfortable using Bisect_ppx `master`, or is it preferable to release this fix to opam soon?

**Comment by liyishuai** [comment]
Since there's no [such](/coq/opam-coq-archive/tree/master/extra-dev) repo for development versions, I'd prefer an OPAM release.

**Comment by aantron** [comment]
Not sure what you mean by a repo. I was referring to pinning with

```
opam pin add bisect_ppx --dev-repo
```

In any case, I'll do a release in the coming days.

**Comment by aantron** [comment]
Just want to confirm that this was merged, as `bisect_ppx` 1.4.0, into opam a couple weeks ago.

**Comment by Chobbes** [comment]
:relaxed:

Thanks for doing this again, this worked the treat :).
